pkg/knuu/instance.go

```diff
@@ -609,6 +610,22 @@ func (i *Instance) SetStartupProbe(startupProbe *v1.Probe) error {
 	return nil
 }
 
+// AddExternalDns sets the external DNS of the instance
+// This function can only be called in the states 'Preparing' and 'Committed'
+func (i *Instance) AddExternalDns(dns string) error {
+	if !i.IsInState(Preparing, Committed) {
+		return fmt.Errorf("adding external DNS is only allowed in state 'Preparing' or 'Committed'. Current state is '%s'", i.state.String())
+	}
+	// check if external DNS is already added
+	for _, d := range i.externalDns {
+		if d == dns {
+			return fmt.Errorf("external DNS '%s' is already added", dns)
+		}
+	}
+	i.externalDns = append(i.externalDns, dns)
+	return nil
+}
```

Comment:
Our of curiosity, is it planned to have removeExternalDNS in the future?

Reply:
I don't see the use case right now. I could add it for completion tho.

Reply:
Agreed with @jrmanes that knuu will not alter an existing environment but rather deploy a new one.